### PR TITLE
Add a type column to the referrals database. #2198

### PR DESCRIPTION
**Pull request by pippinsplugins**
#2198

### CODE REVIEW

File: includes/class-referral-type-registry.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	/**
+	 * Retrieves a type and its associated attributes.
+	 *
+	 * @access public
+	 * @since  2.2
+	 *
+	 * @param string $type_id Type ID.
+	 * @return array|false Array of attributes for the type if registered, otherwise false.
+	 */
+	public function get_type( $type_id ) {
```
Comment (DrewAPicture) [inline includes/class-referral-type-registry.php]: You can skip defining methods that just wrap the parent methods and use magic methods instead. That's the method I'm using for new registries in the reports API, at least.

See [here](https://github.com/easydigitaldownloads/easy-digital-downloads/pull/6273/files#diff-f4ffa2fff58716d72afbcd817bf5d6ebR23) for the `@method` notations. See [here](https://github.com/easydigitaldownloads/easy-digital-downloads/pull/6273/files#diff-f4ffa2fff58716d72afbcd817bf5d6ebR60) for the magic methods.

Just a thought.

Comment (pippinsplugins) [reply includes/class-referral-type-registry.php]: Huh, that's interesting. Obviously both work, what do you think is the benefit to the magic methods?

Comment (DrewAPicture) [reply includes/class-referral-type-registry.php]: Pretty much the only useful aspect of using the magic methods is not defining extra methods for the sole purpose of namespacing the parent functions. If there's a method, it should have a valid purpose, and I'm not sure I consider namespacing to be enough of a reason. Granted, I didn't do it this way previously in AffWP, but that's my thinking nowadays.

Comment (pippinsplugins) [reply includes/class-referral-type-registry.php]: I think for consistency's sake we should either namespace all of them (like AFFWP currently does) or move them all to magic methods.

**Pull request merged**